ui/src/tool-panel/operational-point/operational-point-switches-infobox.tsx

```diff
@@ -0,0 +1,430 @@
+    linkSwitches: (switches: { id: LayoutSwitchId; name: string }[]) => void;
+    unlinkSwitches: (switches: { id: LayoutSwitchId; name: string }[]) => void;
+} {
+    const { t } = useTranslation();
+
+    const [isInitializing, setIsInitializing] = React.useState(true);
+    const [linkingInFlight, addLinkingInFlight, deleteFromLinkingInFlight] =
+        useSetState<LayoutSwitchId>();
+    const [unlinkingInFlight, addUnlinkingInFlight, deleteFromUnlinkingInFlight] =
+        useSetState<LayoutSwitchId>();
+
+    const [preliminaryLinked, addPreliminaryLinked, deletePreliminaryLinked, setPreliminaryLinked] =
+        useSetState<LayoutSwitchId>();
+    const [
+        preliminaryUnlinked,
+        addPreliminaryUnlinked,
+        deletePreliminaryUnlinked,
+        setPreliminaryUnlinked,
+    ] = useSetState<LayoutSwitchId>();
+
+    const [operationalPointSwitches, setOperationalPointSwitches] = React.useState<
+        SwitchWithOperationalPointPolygonInclusions[]
+    >([]);
+
+    const [switchItems, setSwitchItems] = React.useState<LayoutSwitch[]>([]);
+    useRateLimitedTwoPartEffect(
+        async () => {
+            const loadedOperationalPointSwitches = await findOperationalPointSwitches(
+                layoutContext,
+                operationalPoint.id,
+            );
+            const loadedSwitchItems = await getSwitches(
+                loadedOperationalPointSwitches.map(({ switchId }) => switchId),
+                layoutContext,
+            );
+            return { loadedOperationalPointSwitches, loadedSwitchItems };
+        },
+        ({ loadedOperationalPointSwitches, loadedSwitchItems }) => {
+            setIsInitializing(false);
+            setPreliminaryLinked(new Set());
+            setPreliminaryUnlinked(new Set());
+            setOperationalPointSwitches(loadedOperationalPointSwitches);
+            setSwitchItems(loadedSwitchItems);
+        },
+        1000,
+        [
+            layoutContext,
+            operationalPoint.id,
+            getMaxTimestamp(changeTimes.operationalPoints, changeTimes.layoutSwitch),
+        ],
+    );
+
+    const [linkedSwitchItems, unlinkedSwitchItems] = partitionBy(switchItems, (s) => {
+        if (unlinkingInFlight.has(s.id) || preliminaryUnlinked.has(s.id)) {
+            return false;
+        } else if (linkingInFlight.has(s.id) || preliminaryLinked.has(s.id)) {
+            return true;
+        } else {
+            return s.operationalPointId === operationalPoint.id;
+        }
+    });
+
+    const linkSwitches = async (switches: { id: LayoutSwitchId; name: string }[]) => {
+        switches.forEach(({ id }) => {
+            addLinkingInFlight(id);
+            deletePreliminaryUnlinked(id);
+        });
+        try {
+            await Promise.all(
+                switches.map(({ id }) =>
+                    assignOperationalPoint(layoutContext.branch, id, operationalPoint.id),
+                ),
+            );
```

Comment:
Tämä ja tuo alempi tuolla `unlinkSwitches`:issä mielestäni kuulostaisi siltä että näille kannattaisi tehdä bäkkärille massa-asetus-endpoint `Promise.all()`:in sijaan